telemetry.py

```diff
@@ -0,0 +1,158 @@
+        self.surface_altitude = Gauge('surface_altitude', 'surface_altitude in meters')
+        self.altitude = Gauge('altitude', 'altitude in meters')
+        self.apoapsis = Gauge('apoapsis', 'apoapsis in meters')
+        self.velocity = Gauge('velocity', 'velocity in meters per second')
+        self.vertical_vel = Gauge('vertical_vel', 'vertical_vel in meters per second')
+        self.horizontal_vel = Gauge('horizontal_vel', 'horizontal_vel in meters per second')
+        self.periapsis = Gauge('periapsis', 'periapsis in meters')
+
+        # host metrics
+        self.system_cpu_percent = Gauge('system_cpu_percent', 'total host CPU usage in percent')
+
+        # metrics which come from control system calculations, not krpc streams
+        self.gauge_metrics = {}
+        self.counter_metrics = {}
+        self.histogram_metrics = {}
+        self.enum_metrics = {}
+
+        self.gnc_debug = os.environ.get('GNC_DEBUG', None) is not None
+
+
+
+    def start_metrics_server(self, port: int = 8012):
+        # todo some kind of lock or better error handling here if this class is instantiated > once
+        start_http_server(port)
+
+
+    def start_telem_publish(self, telemetry: Telemetry) -> bool:
+        if self.publish_rate_hz == 0:
+            print('not starting telemetry publish thread')
+            return False
+
+        updater_thread = threading.Thread(target=self._telem_publish_thread, args=(telemetry,), daemon=True)
+        updater_thread.start()
+        return True
+
+
+    def _telem_publish_thread(self, telemetry: Telemetry):
+        while True:
+            self.surface_altitude.set(telemetry.surface_altitude())
+            self.altitude.set(telemetry.altitude())
+            self.apoapsis.set(telemetry.apoapsis())
+            self.velocity.set(telemetry.velocity())
+            self.vertical_vel.set(telemetry.vertical_vel())
+            self.horizontal_vel.set(telemetry.horizontal_vel())
+            self.periapsis.set(telemetry.periapsis())
+
+            self.system_cpu_percent.set(psutil.cpu_percent())
+
+            time.sleep(1/self.publish_rate_hz)
+
+    # -------------------------------------------------------------
+    # Worker Thread Function
+    # -------------------------------------------------------------
+    def metric_publisher(self, queue: Queue):
+        """
+        Continuously consume metrics from the queue and update Prometheus metrics.
+        """
+        while True:
+            # Block until an item is available
+            metric_data = queue.get()
+            if metric_data is None:
+                # This can be a signal to gracefully stop if you ever need it
+                break
+
+            name = metric_data["name"]
+            type = metric_data["type"]
+            value = metric_data["value"]
+
+            if type == "gauge":
+                self.gauge_metrics[name].set(value)
+            else:
+                print(f"warning: unsupported metric type {type}")
```

Comment:
would be nice to also support counts for the frame counters at least

dont see any use for supporting histograms